Sources/Workspace/Workspace.swift

```diff
@@ -1636,9 +1628,24 @@ extension Workspace {
 
                             switch extractResult {
                             case .success:
-                                if let expectedPath = self.path(for: artifact), !self.fileSystem.isDirectory(expectedPath) {
-                                    tempDiagnostics.emit(.artifactNotFound(targetName: artifact.targetName, artifactName: expectedPath.basename))
+                                let extractedFiles = tempDiagnostics.wrap { try self.fileSystem.getDirectoryContents(parentDirectory)
+                                    .map{ parentDirectory.appending(component: $0) }
+                                    .filter{ $0 != archivePath }
+                                }
+
+                                guard let artifactPath = extractedFiles?.first(where: { $0.basenameWithoutExt == artifact.targetName }) else {
+                                    return tempDiagnostics.emit(.artifactNotFound(targetName: artifact.targetName, artifactName: artifact.targetName))
                                 }
```

Comment:
👀 key changes here. this is similar logic to the one before by forcing that the binary dependency is in `<package>/<target>` but we do not force specific extension any longer

Reply:
the proviso version `self.path(for: artifact)` use to return `<package>/<target>.xcframework`

Reply:
https://github.com/apple/swift-package-manager/pull/3289/files#diff-134223fe78a1e46f08f8c1e9d166408b7105f941cb2cc4e844ebceb87a8f4ec8L1567